### PR TITLE
Update several examples in documentation to include Puppet examples

### PR DESCRIPTION
**Pull request by dhollinger**
#### Pull Request (PR) description
Add several Puppet DSL examples to the README.

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,27 +0,0 @@
-## 1.1.0 (2017-10-17)
```
Comment (alexjfisher) [inline HISTORY.md:1]: Why is this file being removed? If you think the changelog.md looks better without it, could you include the updated version in this PR?

**Review by kenyon — APPROVED** [review]

**Review by alexjfisher — DISMISSED** [review]
Always great to see docs updated.  I don't think HISTORY.md should be removed though.

File: HISTORY.md (hunk 1 above)
Comment (dhollinger) [reply HISTORY.md:1]: Removing a redundant file as the CHANGELOG.md already fills this role AND has the same content for these versions in there.

Comment (dhollinger) [reply HISTORY.md:1]: CHANGELOG section for reference: https://github.com/voxpupuli/puppet-rsyslog/blob/master/CHANGELOG.md#110-2017-10-17

Comment (alexjfisher) [reply HISTORY.md:1]: It has the same contents *because* of HISTORY.md  It's a feature of github changelog generator.

Comment (alexjfisher) [reply HISTORY.md:1]: See https://github.com/github-changelog-generator/github-changelog-generator#migrating-from-a-manual-changelog

**Review by alexjfisher — APPROVED** [review]

**Pull request merged**